src/ast.rs

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 VMware, Inc.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+use serde_json::Value;
+
+/// A path is a tree of selector nodes.
+///
+/// For example, the JSONPath `$.foo.bar` yields this AST:
+///
+/// ```text
+///              ^
+///             / \
+///            ^   \___ DotName("bar")
+///           / \
+/// Root ___ /   \___ DotName("foo")
+/// ```
+///
+/// A more complicated example: `$.foo[1,2]["bar"]`:
+///
+/// ```text
+///                  ^
+///                 / \
+///                ^   \___ Union
+///               / \            \
+///              ^   \___ Union   \
```

Comment:
Doesm't each `^` correspond to a `Sel`?
```suggestion
///              /   \___ Union   \
```

Reply:
`await fever_gone(); activate_brain()`

Reply:
looking this on an actual laptop helps; accepted, that was a typo indeed